src/WattleScript.Interpreter/Execution/VM/Processor/Processor_InstructionLoop.cs

```diff
@@ -1534,6 +1554,24 @@ private int ExecIndexSet(Instruction i, int instructionPtr)
 
 						return instructionPtr;
 					}
+					case DataType.Range:
+					{
+						h = DynValue.Nil;
+
+						if (idx.String == "to")
+						{
+							obj.Range.To = value.CastToInt() ?? 0;
```

Comment:
Same as ExecRange, this should do a proper type check and throw if you are trying to assign something that can't be coerced to number.

Reply:
Added in [ea86dbd](https://github.com/WattleScript/wattlescript/pull/36/commits/ea86dbd442ce1abc6a2f39118fec572d5ccebfb3)